src/utils/dif_upload.rs

```diff
@@ -655,7 +655,7 @@ fn search_difs(options: &DifUpload) -> Result<Vec<DifMatch<'static>>, Error> {
         }
         walk_difs_directory(base_path, options, |source, name, buffer| {
             debug!("trying to process {}", name);
-            progress.set_message(&name);
+            progress.set_message(name.to_owned());
```

Comment:
Rust Q, do we always prefer `to_owned` over `clone` because it's more generic? (`to_owned` might not clone right? - depends on the trait impl)

Reply:
`clone()` on `&str` returns a `&str`, where in this case `to_owned` will return `String` which we need for `set_message`. I prefer `to_owned` as it's more explicit and obvious what are you want to get out of that call.